### ISSUE TITLE
Bug of card

### ISSUE THREAD

**Issue by jet-snowman** [stripe/stripe-node#290]
Hi there! I have a problem when I tried to create card for some customer. I also tried example on index page and got the same error. 

This is the example:

```node
stripe.customers.create({
  email: 'foo-customer@example.com'
}).then(function(customer){
  return stripe.customers.createSource(customer.id, {
    object: 'card',
    exp_month: 10,
    exp_year: 2018,
    number: '4242 4242 4242 4242',
    cvc: 100
  });
}).then(function(source) {
  return stripe.charges.create({
    amount: 1600,
    currency: 'usd',
    customer: source.customer
  });
}).then(function(charge) {
  // New charge created on a new customer
}).catch(function(err) {
  // Deal with an error
});
```

Response:
```
{ [Error: Missing required param: source.]
  type: 'StripeInvalidRequestError',
  rawType: 'invalid_request_error',
  code: undefined,
  param: 'source',
  message: 'Missing required param: source.',
  detail: undefined,
  raw:
   { type: 'invalid_request_error',
     message: 'Missing required param: source.',
     param: 'source',
     statusCode: 400,
     requestId: 'req_9ofKRcFXZEvl2X' },
  requestId: 'req_9ofKRcFXZEvl2X',
  statusCode: 400 }
```

I've read the documentation and the field source is required but in example and even in test files I don't see this field.

Stripe: 4.14.0
Node js: 5.11
npm: 3.8.6

**Comment by sosnovsky** [comment]
I have the same problem :(

**Comment by fred-stripe** [comment]
Hi @jet-snowman @sosnovsky, what you need to do is pass the `source` as a dictionary as similar to the Stripe API example for creating a token:

https://stripe.com/docs/api/node#create_card_token

```
stripe.customers.create({
  email: 'customer@example.com'
}).then(function(customer){
  return stripe.customers.createSource(customer.id, {
    source: {
        object: 'card',
        exp_month: 10,
        exp_year: 2018,
        number: '4242 4242 4242 4242',
        cvc: 100
     }
  });
}).then(…)
```

See also:
- https://stripe.com/docs/api/node#create_token
- https://stripe.com/docs/api/node#create_card


Unfortunately the example in the README on this repository is incorrect, and Stripe's API example shows how to create a card from a token, not directly from a card number. You should generally be using Stripe.js to tokenize card numbers instead of sending them to the API directly:

**Comment by sosnovsky** [comment]
@fred-stripe thanks for solution!

By the way I fixed error by adding source as object to charges.create:
```
stripe.customers.create({
    email: 'foo-customer@example.com'
  }).then(function(customer) {
    return stripe.charges.create({
      amount: 1600,
      currency: 'usd',
      source: {
        object: 'card',
        exp_month: 10,
        exp_year: 2018,
        number: '4242424242424242',
        cvc: 100
      }
    });
  }).then(...
```

**Comment by fred-stripe** [comment]
Sure, that works too—just keep in mind that charges created that way won't be linked with the customer; you need to pass the "customer" (and optionally the source) when you create the charge. Another way to write this same example:

```javascript
stripe.customers.create({
    email: 'foo-customer@example.com',
    source: {
      object: 'card',
      exp_month: 10,
      exp_year: 2018,
      number: '4242424242424242',
      cvc: 100
    }
  }).then(function(customer) {
    return stripe.charges.create({
      amount: 1600,
      currency: 'usd',
      customer: customer.id
    });
  }).then(...)
```

But again you'd want to use [Stripe.js](https://stripe.com/docs/stripe.js) or [Checkout](https://stripe.com/docs/checkout) to tokenize the card in your live application vs passing the card details directly to the API. Doing so simplifies your PCI compliance, as your app and server won't be handling raw credit card details.

**Comment by sosnovsky** [comment]
@fred-stripe and do you know how I can use card token from my iOS app?
I'm sending token to node api, but always get error "No such token: card_19W...". This token is created by iOS library, so must be working I think, or I'm missing something.

**Comment by fred-stripe** [comment]
That's expected, each card belongs to a Customer, so you need to pass both the customer ID and card or "source" ID when creating a charge:

https://stripe.com/docs/api/node#retrieve_charge

 If you have more questions about integrating your iOS and Node app I'm happy to help, but it may be better to move this discussion to email as this repo is for discussing changes to the stripe-node bindings. :)

You can send an email to Stripe support here:
https://support.stripe.com/email/login

**Comment by sosnovsky** [comment]
@fred-stripe okay, thank you for your help here!

**Comment by dpetrovics-stripe** [comment]
We've got a fix for the README here: #291 , so closing out this issue.
